tests/functionnal/RuleTicket.php

```diff
@@ -787,7 +787,139 @@ public function testTaskTemplateAssignFromRule() {
       $this->string($task_data['content'])->isEqualTo('test content');
    }
 
-   public function testGroupRequesterAssignFromUserGroupsAndRegex() {
+   public function testGroupRequesterAssignFromUserGroupsAndRegexOnUpdateTicketContent() {
+      $this->login();
+
+      // Create rule
+      $ruleticket = new \RuleTicket();
+      $rulecrit   = new \RuleCriteria();
+      $ruleaction = new \RuleAction();
+
+      $ruletid = $ruleticket->add($ruletinput = [
+         'name'         => 'test regex group requester criterion',
+         'match'        => 'AND',
+         'is_active'    => 1,
+         'sub_type'     => 'RuleTicket',
+         'condition'    => \RuleTicket::ONUPDATE,
+         'is_recursive' => 0,
+      ]);
+      $this->checkInput($ruleticket, $ruletid, $ruletinput);
+
+      //create criteria to check if group requester match regex
+      $crit_id = $rulecrit->add($crit_input = [
+         'rules_id'  => $ruletid,
+         'criteria'  => '_groups_id_of_requester',
+         'condition' => \Rule::REGEX_MATCH,
+         'pattern'   => Toolbox::addslashes_deep('/(.+\([^()]*\))/'),   //retrieve groupe with '(' and ')'
+      ]);
+      //change value because of addslashes
+      $crit_input['pattern'] = '/(.+\([^()]*\))/';
```

Comment:
I propose a global solution in #9157. Let's see if it works.

Reply:
it's work with your global solution